src/app/3d/qgs3dviewsmanagerdialog.cpp

```diff
@@ -134,10 +135,11 @@ void Qgs3DViewsManagerDialog::renameClicked()
 
   QgsProject::instance()->viewsManager()->rename3DView( oldTitle, newTitle );
 
-  if ( Qgs3DMapCanvasDockWidget *widget = QgisApp::instance()->get3DMapViewDock( oldTitle ) )
+  if ( QgsDockableWidget *widget = QgisApp::instance()->get3DMapViewDock( oldTitle ) )
   {
     widget->setWindowTitle( newTitle );
-    widget->mapCanvas3D()->setObjectName( newTitle );
+    Qgs3DMapCanvasWidget *canvasWidget = dynamic_cast< Qgs3DMapCanvasWidget * >( widget->widget() );
```

Comment:
With QObject-based classes, it is better to use `qobject_cast` rather than `dynamic_cast`